the-collab-lab/tcl-25-smart-shopping-list
# Implementing accessibility

## Description

This PR focuses on the app accessibility. It ensures that the app is written in semantic HTML. It adds missing attributes to SVG elements and ensures they are given accessible names in the HTML. It also makes sure that accessible names are provided to each ARIA input field . Apart from that, the PR ensures that users can navigate to all interactive interface components of the app  using a keyboard by giving focus styles to those elements that need them. It also makes that all visual information has a  contrast of at least 3:1 against the adjacent background.

## Acceptance Criteria
- [x] Semantically correct HTML
- [x] SVG elements have required attributes and accessible names
- [x] ARIA input fields have accessible names
- [x] Interactive interface components have focus styles
- [x] Visual information has a 3:1 contrast ration against the adjacent background


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin refactoring` and check that branch out with `git checkout refactoring` 
* Then `npm start` to launch the app
* Check that the app is written using semantic HTML
* Check SVG elements have required attributes and accessible names 
* Check ARIA input fields have accessible names
* Check interactive interface components have focus styles
* Check visual information has a 3:1 contrast ration against the adjacent background
